x-pack/plugins/index_lifecycle_management/public/application/sections/policy_table/components/add_policy_to_template_confirm_modal.tsx

```diff
@@ -4,96 +4,55 @@
   EuiFieldText,
   EuiSpacer,
   EuiText,
+  EuiSwitch,
+  EuiButton,
+  EuiComboBoxOptionOption,
 } from '@elastic/eui';
 
 import { PolicyFromES } from '../../../../../common/types';
-import { LearnMoreLink } from '../../edit_policy/components';
-import { addLifecyclePolicyToTemplate, loadIndexTemplates } from '../../../services/api';
+import { addLifecyclePolicyToTemplate, useLoadIndexTemplates } from '../../../services/api';
 import { toasts } from '../../../services/notification';
 import { showApiError } from '../../../services/api_errors';
+import { LearnMoreLink } from '../../edit_policy/components';
 
 interface Props {
   policy: PolicyFromES;
   onCancel: () => void;
 }
-interface State {
-  templates: Array<{ name: string }>;
-  templateName?: string;
-  aliasName?: string;
-  templateError?: string;
-}
-export class AddPolicyToTemplateConfirmModal extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      templates: [],
-    };
-  }
-  async componentDidMount() {
-    const templates = await loadIndexTemplates();
-    this.setState({ templates });
-  }
-  addPolicyToTemplate = async () => {
-    const { policy, onCancel } = this.props;
-    const { templateName, aliasName } = this.state;
-    const policyName = policy.name;
-    if (!templateName) {
-      this.setState({
-        templateError: i18n.translate(
-          'xpack.indexLifecycleMgmt.policyTable.addLifecyclePolicyToTemplateConfirmModal.noTemplateSelectedErrorMessage',
-          { defaultMessage: 'You must select an index template.' }
-        ),
-      });
-      return;
-    }
-    try {
-      await addLifecyclePolicyToTemplate({
-        policyName,
-        templateName,
-        aliasName,
-      });
-      const message = i18n.translate(
-        'xpack.indexLifecycleMgmt.policyTable.addLifecyclePolicyToTemplateConfirmModal.successMessage',
-        {
-          defaultMessage: 'Added policy {policyName} to index template {templateName}',
-          values: { policyName, templateName },
-        }
-      );
-      toasts.addSuccess(message);
-      onCancel();
-    } catch (e) {
-      const title = i18n.translate(
-        'xpack.indexLifecycleMgmt.policyTable.addLifecyclePolicyToTemplateConfirmModal.errorMessage',
-        {
-          defaultMessage: 'Error adding policy "{policyName}" to index template {templateName}',
-          values: { policyName, templateName },
-        }
-      );
-      showApiError(e, title);
-    }
-  };
-  renderTemplateHasPolicyWarning() {
-    const selectedTemplate = this.getSelectedTemplate();
-    const existingPolicyName = get(selectedTemplate, 'settings.index.lifecycle.name');
+
+export const AddPolicyToTemplateConfirmModal: React.FunctionComponent<Props> = ({
+  policy,
+  onCancel,
+}) => {
+  const [isLegacy, setIsLegacy] = useState<boolean>(false);
+  const [templateName, setTemplateName] = useState<string>();
+  const [aliasName, setAliasName] = useState<string>();
```

Comment:
You need to provide an empty string to the field to avoid this error when writing the first character

<img width="529" alt="Screenshot 2020-09-17 at 10 50 45" src="https://user-images.githubusercontent.com/2854616/93454097-c8952d00-f8da-11ea-9d34-de726a87ff08.png">

You might want to strip it out if it's an empty string before sending the form.